core/src/main/java/aima/core/search/basic/informed/RecursiveBestFirstSearch.java

```diff
@@ -122,6 +122,48 @@ public double h(Node<A, S> node) {
 		return h.applyAsDouble(node);
 	}
 
+	public List<A> failure() {
+		return searchController.failure();
+	}
+
+	public List<A> solution(Node<A, S> node) {
+		return searchController.solution(node);
+	}
+	
+	public SuccessorNode newRootNode(S initialState, double pathCost) {
+		return new SuccessorNode(nodeFactory.newRootNode(initialState), this::h);
+	}
+
+	public SuccessorNode newChildNode(Problem<A, S> problem, SuccessorNode node, A action) {
+		return new SuccessorNode(nodeFactory.newChildNode(problem, node.n, action), this::h);
+	}
+	
+	public SuccessorNode getLowestFValueNode(List<SuccessorNode> nodes) {
+		return nodes.stream().min(Comparator.comparingDouble(s -> s.f)).get();
+	}
+
+	public double getSecondLowestFValue(List<SuccessorNode> nodes) {
+		

```

Comment:
you may be able to simplify this logic if you passed the 'best' value as an argument to this function.